store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow.md

```diff
@@ -11,73 +11,346 @@ This tutorial will cover the end-to-end process of creating a bucket, uploading
+
+    ```ts title="src/index.ts // **PLACEHOLDER FOR STEP 1: CHECK MSP HEALTH**"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:check-msp-health'
+    ```
+
+2. Check the health status by running the script:
+
+    ```bash
+    ts-node index.ts
+    ```
+
+    The response should return a **`healthy`** status, like this:
+
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/output-01.html'
+
+## Create a Bucket
 
 Buckets group your files under a specific Main Storage Provider (MSP) and value proposition. Derive a deterministic bucket ID, fetch MSP parameters, then create the bucket. If you run the script multiple times, use a new `bucketName` to avoid a revert, or modify the logic to use your existing bucket in later steps.
 
-In the following code, you will initialize the StorageHub, viem, and Polkadot.js clients on the DataHaven TestNet, sign in via SIWE (Sign in With Ethereum), and pull the MSP’s details/value proposition to prepare for bucket creation. Then you will derive the bucket ID, confirm it doesn’t exist, submit a `createBucket` transaction, wait for confirmation, and finally query the chain to verify that the new bucket’s MSP and owner match our account. 
+In the following code, you will pull the MSP’s details/value proposition to prepare for bucket creation. Then you will derive the bucket ID, confirm it doesn’t exist already, submit a `createBucket` transaction, wait for confirmation, and finally query the chain to verify that the new bucket’s MSP and owner address match the account address that you are using. 
+
+If you'd prefer to step through the steps to create a bucket individually, please see the [Create a Bucket](/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/){target=\_blank} guide.
+
+To create a bucket, you are going to: 
+
+1. Create a `getValueProps` helper method within `mspService.ts`.
+2. Create a `createBucket` helper method within `bucketOperations.ts`.
+3. Update the `index.ts` file to trigger the logic you've implemented.
+4. Check `createBucket` method output
+
+Here are the in-depth instructions:
+
+1. To fetch `valueProps` from the MSP Client, add the following helper function to your `mspService.ts` file:
+
+    ```ts title="src/services/mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts:msp-value-props'
+    ```
+
+2. Add the `getValueProps` method to the export statement at the bottom of the `mspService.ts` file.
+
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts:exports'
+    ```
+
+??? code "View complete `mspService.ts` file"
+
+    ```ts title="src/services/mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts'
+    ```
```

Comment:
This will need to be indented 4 spaces so it shows up under step 2 and so step 3 also renders as a 3 instead of 1